person.rb

```diff
@@ -0,0 +1,33 @@
+require_relative 'nameable'
+
+class Person < Nameable
+  def initialize(age, parent_permission: true, name: 'Unknown')
+    super()
+    @id = Random.rand(1..1000)
+    @name = name
+    @age = age
+    @parent_permission = parent_permission
+    @rentals = []
+  end
+
+  attr_accessor :name, :age
+  attr_reader :id, :rentals
+
+  def add_rental(book, date)
+    Rental.new(date, book, self)
+  end
+
+  private
+
+  def of_age?
+    @age >= 18
+  end
+
+  def can_use_services?
+    @age >= 18 || @parent_permission
+  end
+
```

Comment:
**[OPTIONAL BUT HIGHLY RECOMMENDED]**

- [ ] It would be great if you make your method such as below to make it more concise to make your app more efficient 👍

```ruby
  def can_use_services?
    of_age? || @parent_permission
  end
```